src/sentry/sentry_metrics/consumers/indexer/slicing_router.py

```diff
@@ -0,0 +1,80 @@
+from typing import MutableMapping, Optional
+
+from arroyo import Message, Topic
+from confluent_kafka import Producer
+from django.conf import settings
+
+from sentry.ingest.partitioning import (
+    Sliceable,
+    is_sliced,
+    map_logical_partition_to_slice,
+    map_org_id_to_logical_partition,
+)
+from sentry.sentry_metrics.consumers.indexer.routing_producer import (
+    MessageRoute,
+    MessageRouter,
+    RoutingPayload,
+)
+from sentry.utils import kafka_config
+
+
+class SlicingRouter(MessageRouter):
```

Comment:
The fact that this class behaves differently in the sliced and unsliced version was very unexpected to me. Part of it might relate to inconsistencies in how settings are provided between sliced and unsliced.

For example an output_topic is passed (required) then it gets used only if the topic is not sliced and ignored if it is. If it is sliced it goes to settings but not if unsliced.

I think this needs to be consistent in both cases. Either always pass the topics in, or always go to settings, the hybrid of the two is confusing.

Reply:
The original intention was to have the SlicingRouter work as an unsliced router as well. There is nothing which prevents a Slicing router to have a single output topic and send data to it. Hence, the need for passing the output_topic. I can change that to always rely on settings and work with sliced configuration only.